### PR TITLE
Remove MapperService and related components

### PR DESCRIPTION
**Pull request by mfussenegger**
Follow up to https://github.com/crate/crate/pull/16093

(The field mappers are now also pretty much unused, but I'll defer the removal of them into a separate PR as this is already quite big)

### CODE REVIEW

**Review by BaurzhanSakhariev — APPROVED** [review]

File: server/src/main/java/io/crate/metadata/DanglingArtifactsService.java [hunk 1]
```diff
@@ -70,12 +62,11 @@ protected void doClose() {
     public void clusterChanged(ClusterChangedEvent event) {
         if (LOGGER.isInfoEnabled() && event.isNewCluster()) {
             for (ObjectCursor<String> key : event.state().metadata().indices().keys()) {
-                for (Pattern pattern : danglingPatterns) {
```
Comment (romseygeek) [inline server/src/main/java/io/crate/metadata/DanglingArtifactsService.java:73]: Is this supposed to be part of this PR?

File: server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java [hunk 2]
```diff
@@ -19,151 +19,19 @@
...
-    public DocumentFieldMappers mappers() {
-        return this.fieldMappers;
-    }
-
-    public Map<String, ObjectMapper> objectMappers() {
-        return this.objectMappers;
+    private DocumentMapper() {
     }
 
     public static ParsedDocument createDeleteTombstoneDoc(String index, String id) throws MapperParsingException {
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java:37]: We could move these static methods to ParsedDocument and delete DocumentMapper entirely?

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 3]
```diff
@@ -1084,11 +1066,7 @@ interface IndexDeletionAllowedPredicate {
      * Returns true if the provided field is a registered metadata field (including ones registered via plugins), false otherwise.
      */
     public boolean isMetadataField(String field) {
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:1068]: It looks as though this method is unused so we can just delete it?

**Review by romseygeek — APPROVED** [review]
Nice!  I left a couple of comments but no need for another review.

File: server/src/main/java/io/crate/metadata/DanglingArtifactsService.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/metadata/DanglingArtifactsService.java:73]: Good point, I'll extract it into a separate PR. My first iteration had looked a bit different and required changes here

**Comment by mfussenegger** [comment]
retest this please


flaky swap table tests

**Comment by mfussenegger** [comment]
retest this please

pretty please?

**Pull request merged**